src/view/renderer.js

```diff
@@ -473,19 +536,92 @@ export default class Renderer {
 	 * filler should be rendered.
 	 */
 	_updateChildren( viewElement, options ) {
+		const diff = this._diffElementChildren( viewElement,
+			{ inlineFillerPosition: options.inlineFillerPosition, bind: true, withChildren: true } );
+
+		if ( diff ) {
```

Comment:
Again – early return.

And – how is it possible that we call this method when its children are up to date?

Reply:
> And – how is it possible that we call this method when its children are up to date?

I'm not sure what you meant here? Do you have any particular case in mind?

Reply:
I meant the logic of this method – you call `updateChildren()` based on `markedElements`. My assumption is – if an element is marked its children have changed. And hence, the `diff` should not be empty in this case.

The existence of this `if()` here means that my assumption is wrong. But why? We need to have a clarity on:

1. Why is this method called for up-to-date element in the first place.
2. Can we prevent calling it for up-to-date element (so to remove this `if()`).
3. Eventually, if it's most convenient to have this `if()` here and it's really needed (there are no other straightforward solutions), we need a comment.

Reply:
One thing, even in situations when children are equal we [mark descendant text nodes to sync](https://github.com/ckeditor/ckeditor5-engine/pull/1424/files#diff-4c87133ed830fc4ea0646426deba32cfR561) so I'm not sure if we may completely skip such situations (if that's what you mean).

Reply:
Didn't noticed your comment earlier @Reinmar, so my last comment refers to previous ones.

As for the latest comment - `diff` is empty not when elements are up-to-date (then you will have `diff` result like `[ equal, equal, ... ]`), but when the `viewElement` which is passed to `updateChildren()` and then `diffChildren()` does not have corresponding DOM element (it cannot be mapped which usually means that corresponding DOM element was removed from DOM, but probably also similar cases as with attributes - https://github.com/ckeditor/ckeditor5-engine/pull/1424#discussion_r195065595). This is not a new `if()` here, as [it was there before](https://github.com/ckeditor/ckeditor5-engine/blob/3e74554fa97741189050f7a4014ee6c558645cb8/src/view/renderer.js#L479) and it was just moved to `diffChildren()` method together with comment explaining what happened - https://github.com/ckeditor/ckeditor5-engine/pull/1424/files#diff-4c87133ed830fc4ea0646426deba32cfR611.